app/src/main/java/com/awscherb/cardkeeper/data/handler/BaseRealmHandler.java

```diff
@@ -84,16 +67,67 @@
     //================================================================================
 
     @Override
-    public Observable<Void> deleteObject(T object) {
+    public Completable deleteObject(T object) {
         Realm realm = Realm.getDefaultInstance();
         return doAsync(object, realm, o ->
                 realm.where(getModelClass())
                         .equalTo("id", o.getId())
                         .findAll()
                         .first()
-                        .deleteFromRealm())
-                .map(delete -> null);
+                        .deleteFromRealm()).toCompletable();
+    }
+
+    //================================================================================
+    // Helper
+    //================================================================================
+
+    private Single<T> doAsync(T object, Realm realm, Consumer<T> consumer) {
+        return Single.create(e -> {
```

Comment:
It looks like if there's an error, it calls `e.onError()` in catch and then `e.onSuccess(object);` at the end 😄  So there's indeed a mistake in the contract.